tsdb/_tools/dedupe.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+####################################################################
+#
+# A tool that dedupes a sorted anonymized metricbeat dump.
+#
+####################################################################
+#
+# Expects sorted anonymized metricbeat dump as input via standard
+# in and returns a deduped sorted anonymized metric beat output via
+# standard out. Also seperately generates 'dupes-' prefixed files 
+# per metric set name containing the dupes for manual inspection.
+#
+####################################################################
+
+import json
+import sys
+
+def generate_event_key(parsed_line):
+    return parsed_line['kubernetes']['event']['involved_object']['uid']
+
+def generate_state_container_key(parsed_line):
+    key = parsed_line['kubernetes']['container']['name']
+    key += parsed_line['kubernetes']['pod']['name']
+    key += parsed_line['kubernetes']['node']['name']
+    container_id = parsed_line.get('kubernetes',{}).get('container', {}).get('id')
+    if (container_id is not None):
+        key += container_id
+    return key
+
+def generate_state_pod_key(parsed_line):
+    return parsed_line['kubernetes']['pod']['name'] + parsed_line['kubernetes']['node']['name']
+
+def generate_container_key(parsed_line):
+    return parsed_line['kubernetes']['container']['name'] + parsed_line['kubernetes']['pod']['name'] + parsed_line['kubernetes']['node']['name']
+
+def generate_volume_key(parsed_line):
+    return parsed_line['kubernetes']['volume']['name'] + parsed_line['kubernetes']['pod']['name'] + parsed_line['kubernetes']['node']['name']
+
+def generate_pod_key(parsed_line):
+    return parsed_line['kubernetes']['pod']['name'] + parsed_line['kubernetes']['node']['name']
+
+def generate_node_key(parsed_line):
```

Comment:
A minor thing, but this function and `generate_state_node_key` are identical. There's also several places where you could replace `parsed_line['kubernetes']['node']['name']` with `generate_state_node_key(parsed_line)` just to cut down on duplication if you'd like. Not a big deal, though.